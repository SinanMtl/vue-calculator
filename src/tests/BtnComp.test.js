import { describe, expect, it } from 'vitest';
import { mount } from '@vue/test-utils';
import Btn from '../components/BtnComp.vue';

describe('Button component', () => {
  const wrapper = mount(Btn);
  
  it('should be mounted', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should be clicked', () => {
    wrapper.vm.$emit('click', 123);

    const emits = wrapper.emitted();

    expect(emits.click[0][0]).toBe(123)
  });
})

