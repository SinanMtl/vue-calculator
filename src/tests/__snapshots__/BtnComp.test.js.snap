// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > should be mounted 1`] = `"<button class=\\"w-20 h-20 bg-gray-500 text-white text-5xl rounded-full hover:bg-gray-800 font-inter inline-flex items-center justify-center\\"></button>"`;
